### DISCUSSION TITLE
Failed to provide a binding for b

### DISCUSSION THREAD

**Discussion by white-gecko** [comunica/comunica#1405] [Q&A]
When I try to execute a select query that leaves some variables unbound in some of the bindings I get the error:

Uncaught (in promise) Error: The endpoint http://localhost:5000/sparql failed to provide a binding for b.

While I would expect that there is not binding for a variable in some cases. That is, why there is the `bindings.has('b')` method.

This should be me MWE code:

```
    let graph = "http://example.org/"
    let datasetClause = `from <${graph}>`

    const queryString = `select ?a ?b ${datasetClause} where { { ?a ?p ?o } union {?b ?p ?o } }`

    let queryEndpoint = "http://localhost:5000/sparql"
    let queryEngine = new QueryEngine()
    let sources = [{ type: 'sparql', value: queryEndpoint }]

    const bindingsStream = await queryEngine.queryBindings(queryString, {
      sources: sources
    })

    console.log(bindingsStream)
    console.log(Object.prototype.toString.call(bindingsStream))
    console.log(await bindingsStream.toArray())
```

Is this a bug or am I doing something wrong?

**Comment by rubensworks** [comment]
This is definitely a bug on our end. I've created #1406 to track it.
